newIDE/app/src/InstancesEditor/index.js

```diff
@@ -218,12 +218,16 @@ export default class InstancesEditor extends Component<Props> {
       'mouseup',
       this.keyboardShortcuts.onMouseUp
     );
-    this.pixiRenderer.view.addEventListener('mousemove', event =>
-      this.props.onMouseMove(event)
-    );
+    this.pixiRenderer.view.addEventListener('mousemove', event => {
+      this.props.onMouseMove(event);
+    });
     this.pixiRenderer.view.addEventListener('mouseout', event => {
+      this.fpsLimiter.updateMaxFps(15); // Reduce the FPS when the mouse is out of the window.
```

Comment:
Two things:
- Use pointerout for the FPS limitation, so that it works with pencils.
- Is this working with touchscreens? There are no way to know if a touch is "out", so for touchscreens we might need something where a touch is setting FPS to 60 (I think "pointerover" should work for this for both mouse, pencils and otuch). Then a timeout that reset the fps to 15 4-5 seconds later. If in the meantime, we receive a mousemove/pointermouve/touchmove/touchstart/mouseover/etc (I let you judge what we need), then we reset the time of last interaction.

Reply:
If you do this, I suggest to move the responsibility to the fpsLimiter:

- Add a `notifyInteractionHappened()` method. When called, it set the "timeOfLastInteraction" to performance.now() and immediately set the fps to 60.
- In the rendering of the scene, call a `fpsLimiter.update()` or something that will be called regularly. If performance.now() - timeOfLastInteraction > 4 seconds => reduce to 15 fps.
- That's it???

Reply:
I've updated the logic to handle this, I've added the notification on most "notable interactions" (drag/drop/resize/move/scroll/zoom) but not all of them (just a click or a mouse move doesn't)
I've tested as much as I could and it looks fine (see video)